### DISCUSSION TITLE
Ottai Calibration

### DISCUSSION THREAD

**Discussion by dmitrievichh** [NightscoutFoundation/xDrip#3760] [Bug reports] [answered]
Hello! Very glad on ability of ottai connection. But I can't make calibration getting some errors.
![Screenshot_20241106_113054](https://github.com/user-attachments/assets/837e67f6-8c10-43dc-9227-9835ab193353)
Is calibrations not available for now or I'm doing something wrong?
I have last night xdrip version of 03.11.2024.

**Comment by parapenT1sta** [comment]
It seems the same error I got using L2: https://github.com/NightscoutFoundation/xDrip/discussions/3723

**Comment by Navid200** [comment]
Ottai has been added to the companion app list.
When you use a companion app, you cannot calibrate through xDrip.  You can calibrate using the companion app unless the companion app does not allow you to calibrate.

So, in this case, you have installed the Ottai app on your phone as well as xDrip.  Can you not use the Ottai app to calibrate?.

**Reply by dmitrievichh** [reply]
Unfortunately, Ottai app doesn't have calibration functionality :(

**Reply by Navid200** [reply, marked as answer]
I see.
If the manufacturer believes that there is no need for calibration, it will be dangerous for xDrip to offer that in companion app mode.
It could lead to serious errors.   

The xDrip logic has been to offer solutions to diabetics to help them when possible with minimal stepping into the shoes of manufacturers.  

I personally do what I can to improve xDrip.  This one is something that raises questions for me and I cannot think of a way I can help.  I am sorry.

**Reply by dmitrievichh** [reply]
Thanks for answering!

**Reply by parapenT1sta** [reply]
> I see. If the manufacturer believes that there is no need for calibration, it will be dangerous for xDrip to offer that in companion app mode. It could lead to serious errors.
> 
> The xDrip logic has been to offer solutions to diabetics to help them when possible with minimal stepping into the shoes of manufacturers.
> 
> I personally do what I can to improve xDrip. This one is something that raises questions for me and I cannot think of a way I can help. I am sorry.

In fact I think we should add calibrations because for example, none of Libre official apps or readers allow calibrations and we use xDrip because we believe that Libre with calibrations is much better.

**Reply by Navid200** [reply]
My worry is problems that could be caused by unnecessary calibrations.

There was a time calibration was necessary.  Dexcom G5 was an example.  Without calibration, a G5 would drift significantly because G5 had no means of addressing the impact of the immune system on the sensitivity of the sensor to glucose.

This all changed when G6 came out.  But, there are individuals who insist to calibrate even G6.
Dexcom advises against unnecessary calibrations.  But, it allows to calibrate.  xDrip respects that and passes calibrations to a G6.

If a manufacturer does not allow calibration and we encourage users to calibrate and something goes wrong, who will be at fault?  Of course, many are going to say the user.  But, if some day push comes to shove and some of us are taken to court, no one will be there to say that then!

**Reply by Navid200** [reply]
If blood glucose is low and because of a faulty calibration, the CGM shows that glucose is high, the user may take more insulin instead of taking carbs.  This could cause severe hypoglycemia or even worse.   This is why I would like us to tread carefully with respect to calibrations.

**Comment by RogenBenastra** [comment]
But some Ottai semsors make a barefaced lie. Instead of displaying 5 on a glucometer, they show 8. What shall we do with these readings? Can we correct them anyway?
